src/agent/rustjail/src/wasm.rs

```diff
@@ -0,0 +1,155 @@
+// Copyright (c) 2022 Ant Group
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+macro_rules! cfg_arch_support_wasm {
+    ($($item:item)*) => {
+        $(
+            #[cfg(any(
+                target_arch = "x86_64",
+                target_arch = "x86",
+                target_arch = "aarch64",
+                target_arch = "arm",
+                target_arch = "riscv64",
+                target_arch = "riscv32"
+            ))]
+            $item
+        )*
+    }
+}
+
+macro_rules! cfg_arch_not_support_wasm {
+    ($($item:item)*) => {
+        $(
+            #[cfg(not(any(
+                target_arch = "x86_64",
+                target_arch = "x86",
+                target_arch = "aarch64",
+                target_arch = "arm",
+                target_arch = "riscv64",
+                target_arch = "riscv32"
+            )))]
+            $item
+        )*
+    }
+}
+
+use std::process::exit;
+
+use crate::container::CLOG_FD;
+use crate::log_child;
+use crate::sync::write_count;
+
+cfg_arch_support_wasm! {
+    use anyhow::Result;
+    use std::os::unix::io::RawFd;
+
+    #[cfg(feature = "standard-oci-runtime")]
+    use crate::console;
```

Comment:
Are we running wasm containers in runk mode as well?

Reply:
I didn't test with runk, but I think they can work together.